projects/igniteui-angular/src/lib/grids/grid/grid-filtering-advanced.spec.ts

```diff
@@ -2478,293 +2486,160 @@ describe('IgxGrid - Advanced Filtering #grid', () => {
                 fix.detectChanges();
 
                 // Press 'Enter' on the root group's operator line
-                // and verify that the root group and all of its children become selected.
                 let rootOperatorLine = GridFunctions.getAdvancedFilteringTreeRootGroupOperatorLine(fix);
-                UIInteractions.simulateKeyDownEvent(rootOperatorLine, 'Enter');
-                fix.detectChanges();
-                await wait(200);
-                fix.detectChanges();
-                await wait(200);
-                fix.detectChanges();
+                const keyboardEvent = new KeyboardEvent('keydown', { key: 'Enter' });
+                rootOperatorLine.dispatchEvent(keyboardEvent);
```

Comment:
You can use   UIInteractions.triggerEventHandlerKeyDown('Enter ', rootOperatorLine);

Reply:
I can't use this method here since the rootOperatorLine doesn't have DebugElement.